Add instructions for pre-build images with Github Actions (#614)
astronomer/docs: astro/ci-cd.md

            astrocloud deploy ${{ secrets.PROD_ASTRONOMER_DEPLOYMENT_ID }}
    ```

### GitHub Actions (With Pre-Build Base Image)

The process for consuming secrets during the build of images with dependancies in private repositories is described here: [Install Python Packages from a Private GitHub Repository](develop-project.md). Special considerations have to be made for setting up GitHub Actions workflow to build and deploy these images.

This setup assumes the following prerequisites:

- You have completed the instructions to [Install Python Packages from a Private GitHub Repository](develop-project.md)

> - The Private Key file that was used to authenticate to GitHub is still available.

Author: You prefer "The Private Key file that _you_ used..." over "The Private Key file that _was_ used"?

            astrocloud deploy ${{ secrets.PROD_ASTRONOMER_DEPLOYMENT_ID }}
    ```

</TabItem>

<TabItem value="custom">

When building images that consume secrets or build-time arguments, special considerations have to be made for setting up GitHub Actions workflows. The images are build in two stages to allow extra options to be set during the first build. Some examples of custom images that are built in two stages can be found in [Install Python Packages from Private Sources](develop-project.md#install-python-packages-from-private-sources).

> This setup assumes the following prerequisites:

Reviewer: ```suggestion
#### Prerequisites
```

> - You have completed the instructions to [Install Python Packages from Private Sources](develop-project.md#install-python-packages-from-private-sources) and can build the image locally.
> - If installing from a private GitHub Repo, you have access to the private key file used to authenticate to GitHub.
> 
> OR
> 
> - If installing from a private PyPi Index, you have a username and password to authenticate to the index.

Reviewer: ```suggestion
To complete this setup, you need:

- An Astro project with a custom Runtime image. For an example implementation of this, see [Install Python Packages from Private Sources](develop-project.md#install-python-packages-from-private-sources).
```

    ```yaml
    name: Astronomer CI - Custom base image

    on:
      push:
        branches:
          - main

    jobs:
      build:
        runs-on: ubuntu-latest
        env:
          ASTRONOMER_KEY_ID: ${{ secrets.ASTRO_ACCESS_KEY_ID_DEV }}
          ASTRONOMER_KEY_SECRET: ${{ secrets.ASTRO_SECRET_ACCESS_KEY_DEV }}
        steps:
        - name: Check out the repo
          uses: actions/checkout@v2
    ```

    If installing from a private GitHub Repo, add the following steps:
    ```yaml
        - name: Create SSH Socket
          uses: webfactory/ssh-agent@v0.5.4
          with:
            ssh-private-key: ${{ secrets.GITHUB_SSH_KEY }}
        - name: (Optional) Test SSH Connection - Should print hello message.
          run: (ssh git@github.com) || true
        - name: Build Dockerfile.build image
          uses: docker/build-push-action@v2
          with:
            tags: custom-<astro-runtime-image>
            load: true
            file: Dockerfile.build
            ssh: |
              github=${{ env.SSH_AUTH_SOCK }}
        - name: Deploy to Astro
          run: |

>             brew install astronomer/cloud/astrocloud
>             astrocloud deploy ${{ secrets.ASTRONOMER_DEPLOYMENT_ID }}

Reviewer: ```suggestion
            brew install astronomer/tap/astro
            astro deploy ${{ secrets.ASTRONOMER_DEPLOYMENT_ID }}
```

    ```

    If installing from a private PyPi Index, add the following steps:
    ```yaml
        - name: Build Dockerfile.build image
          uses: docker/build-push-action@v2
          with:
            tags: custom-<astro-runtime-image>
            load: true
            file: Dockerfile.build
            build-args: |
              PIP_EXTRA_INDEX_URL=${{ secrets.PIP_EXTRA_INDEX_URL }}
        - name: Deploy to Astro
          run: |

>             brew install astronomer/cloud/astrocloud
>             astrocloud deploy ${{ secrets.ASTRONOMER_DEPLOYMENT_ID }}

Reviewer: ```suggestion
            brew install astronomer/tap/astro
            astro deploy ${{ secrets.ASTRONOMER_DEPLOYMENT_ID }}
```
